fastpool/src/bounded.rs

```diff
@@ -195,6 +195,51 @@ impl<M: ManageObject> Pool<M> {
         })
     }
 
+    /// Replenishes the pool with at most `most` number of new objects.
+    ///
+    /// * Returns `Ok(n)` if successfully replenished `n` objects.
+    /// * Returns `Ok(0)` if no replenishment is needed.
+    /// * Returns `Err(e)` if an error occurred while replenishing.
+    pub async fn replenish(&self, mut most: usize) -> Result<usize, M::Error> {
+        let permit = loop {
+            if most == 0 {
+                return Ok(0);
+            }
+
+            match self.permits.try_acquire(most) {
+                Some(permit) => break permit,
+                None => {
+                    most = most.min(self.permits.available_permits());
+                    continue;
+                }
+            }
+        };
+
+        most = most.saturating_sub(self.slots.lock().deque.len());
+        if most == 0 {
+            return Ok(0);
+        }
```

Comment:
Maybe release `n` permits where `n = self.slots.lock().deque.len()` so that other gets won't be blocked.

Reply:
Requires `mea` to support partially release permits in a `SemaphorePermit`.